src/Tizen.Applications.Cion/Tizen.Applications/ServerBase.cs

```diff
@@ -0,0 +1,399 @@
+            if (ret != Interop.Cion.ErrorCode.None)
+            {
+                throw CionErrorFactory.GetException(ret, "Failed to stop server.");
+            }
+        }
+
+        /// <summary>
+        /// Disconnects with the peer.
+        /// </summary>
+        /// <param name="peerInfo">The peer to disconnect.</param>
+        /// <exception cref="ArgumentException">Thrown when the given peer info is invalid.</exception>
+        /// <since_tizen> 9 </since_tizen>
+        public void Disconnect(PeerInfo peerInfo)
+        {
+            Interop.Cion.ErrorCode ret = Interop.CionServer.CionServerDisconnect(_handle, peerInfo?._handle);
+            if (ret != Interop.Cion.ErrorCode.None)
+            {
+                throw CionErrorFactory.GetException(ret, "Failed to stop server.");
+            }
+        }
+
+        /// <summary>
+        /// Sends the payload to a peer asynchronously.
+        /// </summary>
+        /// <param name="payload">The payload to send.</param>
+        /// <param name="peerInfo">The peer to send payload.</param>
+        /// <exception cref="ArgumentException">Thrown when the payload is not valid.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when there is no such connected cion client.</exception>
+        /// <since_tizen> 9 </since_tizen>
+        public Task<PayloadAsyncResult> SendPayloadAsync(Payload payload, PeerInfo peerInfo)
+        {
+            if (payload?.Id.Length == 0 || peerInfo?.UUID.Length == 0)
+            {
+                throw new ArgumentException("Payload or peerinfo is invalid.");
+            }
+
+            TaskCompletionSource<PayloadAsyncResult> tcs = new TaskCompletionSource<PayloadAsyncResult>();
+            _tcsDictionary[Tuple.Create(payload.Id, peerInfo.UUID)] = tcs;
+
+            if (_payloadAsyncResultCb == null)
+            {
+                Interop.CionServer.CionServerPayloadAsyncResultCb cb = new Interop.CionServer.CionServerPayloadAsyncResultCb(
+                    (IntPtr result, IntPtr userData) =>
+                    {
+                        PayloadAsyncResult resultPayload = new PayloadAsyncResult(new PayloadAsyncResultSafeHandle(result, false));
+                        TaskCompletionSource<PayloadAsyncResult> tcsToReturn = _tcsDictionary[Tuple.Create(resultPayload.PayloadId, resultPayload.PeerInfo.UUID)];
+                        _tcsDictionary.Remove(Tuple.Create(resultPayload.PayloadId, resultPayload.PeerInfo.UUID));
+                        Interop.Cion.ErrorCode clone_ret = Interop.CionPayloadAsyncResult.CionPayloadAsyncResultClone(result, out PayloadAsyncResultSafeHandle clone);
+                        if (clone_ret != Interop.Cion.ErrorCode.None)
+                        {
+                            Log.Error(LogTag, "Failed to clone result.");
+                            tcsToReturn.SetResult(new PayloadAsyncResult());
```

Comment:
에러 전파를 위해서는 `SetException`을 사용하시면 됩니다.
```suggestion
                            tcsToReturn.SetException(CionErrorFactory.GetException(clone_ret, "Failed to send payload"););
```